osquery/core/http.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/core.h>
+#include <osquery/database.h>
+#include <osquery/logger.h>
+#include "osquery/core/http.h"
+
+#include <curl/curl.h>
+
+namespace osquery{
+
+	std::string decodeError(HTTPResponse resp){
+		long code = resp.getReturnCode();
+
+		switch(code){
```

Comment:
We do some hefty error code manipulation here for SQLite: https://github.com/facebook/osquery/blob/master/osquery/sql/sqlite_util.cpp

It would probably be worth following a similar pattern.